data/src/main/java/com/onboarding/data/service/RickAndMortyRepositoryImpl.kt

```diff
@@ -3,24 +3,33 @@ package com.onboarding.data.service
 import com.onboarding.data.mapper.transform
 import com.onboarding.data.service.api.RickAndMortyApi
 import com.onboarding.data.service.response.CharacterBaseResponse
+import com.onboarding.data.service.response.EpisodeBaseResponse
 import com.onboarding.domain.entity.CharacterBase
+import com.onboarding.domain.entity.EpisodeBase
 import com.onboarding.domain.usecase.RickAndMortyRepository
 import com.onboarding.domain.util.Result
 
 class RickAndMortyRepositoryImpl : RickAndMortyRepository {
-    private val generator = RickAndMortyRequestGenerator()
+    private val callResponse = RickAndMortyRequestGenerator().createService(RickAndMortyApi::class.java)
 
     override fun getAllCharacters(): Result<CharacterBase> {
-        val callResponse = generator.createService(RickAndMortyApi::class.java)
-            .callAllCharacters()
         return try {
-            val response = callResponse.execute()
+            val response = callResponse.callAllCharacters().execute()
             Result.Success((response.body() as CharacterBaseResponse).transform())
         } catch (e: Exception) {
             Result.Failure(Exception(EXCEPTION_MESSAGE))
         }
     }
 
+    override fun getAllEpisodes(): Result<EpisodeBase> {
+        return try {
+            val response = callResponse.callAllEpisodes().execute()
+            Result.Success((response.body() as EpisodeBaseResponse).transform())
```

Comment:
here you should check if the response is successful before setting the result. Same in the getAllCharacters method

Reply:
done